### PR TITLE
Add readme for the copy handler component

### PR DESCRIPTION
**Pull request by JustinyAhin**
## Description
Added documentation for the copy handler component (see #22891)

## How has this been tested?
N/A, documentation only

## Types of changes
Documentation

## Checklist:
- [n/a] My code is tested.
- [n/a] My code follows the WordPress code style. 
- [n/a] My code follows the accessibility standards. 
- [n/a] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/block-editor/src/components/copy-handler/README.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+);
+```
+
+_Note:_ The `CopyHandler` only catch cut/copy events coming from its props.children. In this example, the child is the `BlockList` component.
+
+### Props
+
+### `children`
+
+-   **Type:** `Object`
```
Comment (ntsekouras) [inline packages/block-editor/src/components/copy-handler/README.md:36]:
```suggestion
-   Type: `Element`
```

File: packages/block-editor/src/components/copy-handler/README.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+An object that contains the component children.
```
Comment (ntsekouras) [inline packages/block-editor/src/components/copy-handler/README.md:38]:
```suggestion
Children are passed as children of `CopyHandler `.
```

File: packages/block-editor/src/components/copy-handler/README.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
+# Copy Handler
+
+The `CopyHandler` component handles the behaviors of the editor when the user copies or cuts a block or a selection of blocks.
```
Comment (ntsekouras) [inline packages/block-editor/src/components/copy-handler/README.md:3]: Maybe something more like:

> The `CopyHandler` component handles the `copy`, `cut` and `paste` events of its children blocks. It handles multiple selection of blocks as well.

File: packages/block-editor/src/components/copy-handler/README.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+Concretely, it handles the display of success messages and takes care of copying the block to the clipboard using the [raw handling API](https://github.com/WordPress/gutenberg/tree/master/packages/blocks/src/api/raw-handling).
```
Comment (ntsekouras) [inline packages/block-editor/src/components/copy-handler/README.md:5]: @mcsf can you please take a look here?

File: packages/block-editor/src/components/copy-handler/README.md [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+import { BlockList, CopyHandler } from '@wordpress/block-editor';
+
+const MyCopyHandler = () => (
+	<CopyHandler>
+		<BlockList />
+	</CopyHandler>
+);
+```
+
+_Note:_ The `CopyHandler` only catch cut/copy events coming from its props.children. In this example, the child is the `BlockList` component.
```
Comment (ntsekouras) [inline packages/block-editor/src/components/copy-handler/README.md:30]: The events that are handled are three: cut, copy, paste. I think this should be just a part of the main description.

**Review by ntsekouras — COMMENTED** [review]
Hey @JustinyAhin - thanks for all your work on this and all the other work I've seen from you. It's really helpful to improve documentation!

I left some small comments but most importantly I pinged @mcsf who had worked on this to take a look about the raw API you're mentioning.

File: packages/block-editor/src/components/copy-handler/README.md [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+import { BlockList, CopyHandler } from '@wordpress/block-editor';
+
+const MyCopyHandler = () => (
+	<CopyHandler>
+		<BlockList />
+	</CopyHandler>
+);
+```
+
+_Note:_ The `CopyHandler` only catch cut, copy and paste events coming from its props.children. In this example, the child is the `BlockList` component.
```
Comment (ntsekouras) [inline packages/block-editor/src/components/copy-handler/README.md:30]: >only catch cut,

Change to `catches`

File: packages/block-editor/src/components/copy-handler/README.md [hunk 7]
```diff
@@ -0,0 +1,42 @@
+# Copy Handler
+
+The `CopyHandler` component handles the copy, cut and paste events of its children blocks. It handles multiple selection of blocks as well.
+
+Concretely, it handles the display of success messages and takes care of copying the block to the clipboard using the [raw handling API](https://github.com/WordPress/gutenberg/tree/master/packages/blocks/src/api/raw-handling).
```
Comment (ntsekouras) [inline packages/block-editor/src/components/copy-handler/README.md:5]: >takes care of copying the block

Can you change this to `block/blocks`?

**Review by ntsekouras — COMMENTED** [review]
This looks good! 👍 - I left two super minor details to change and I pinged again @mcsf to check/verify the Raw handling reference.

File: packages/block-editor/src/components/copy-handler/README.md (hunk 7 above)
Comment (JustinyAhin) [reply packages/block-editor/src/components/copy-handler/README.md:5]: You mean change "takes care of copying the block" to "takes care of copying the block/blocks"?

**Comment by JustinyAhin** [comment]
@ntsekouras I'm also wondering why the tests are also failing here. I rebase and force pushed but it doesn't seem to fix.

**Comment by ntsekouras** [comment]
>@ntsekouras I'm also wondering why the tests are also failing here. I rebase and force pushed but it doesn't seem to fix.

It seems to be package.json out of sync on master. It will be fixed and then you'll have to rebase again. I'll ping you when is okay.

**Comment by ntsekouras** [comment]
You can rebase now. Fixed here: https://github.com/WordPress/gutenberg/pull/25489

File: packages/block-editor/src/components/copy-handler/README.md [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+
+![Copy cut behaviours](https://make.wordpress.org/core/files/2020/09/copy-cut-behaviour.gif)
```
Comment (mcsf) [inline packages/block-editor/src/components/copy-handler/README.md:7]: Two notes:

* The GIF itself doesn't clearly show what is meant to happen. Perhaps [this one from the original pull request](https://user-images.githubusercontent.com/150562/81698101-6e341d80-945d-11ea-9bfb-b20781f55033.gif) would be better.

* _Copy cut behaviours_ should be _Copy/cut behaviours_ or _Copy and cut behaviours_.

File: packages/block-editor/src/components/copy-handler/README.md (hunk 7 above)
Comment (mcsf) [inline packages/block-editor/src/components/copy-handler/README.md:5]: > copying the block to the clipboard using the raw handling API

Actually, the raw handling API is only used with _paste_ events (`pasteHandler`).

In contrast, cut/copy actions require what is essentially the opposite of the raw handling API: the `serialize` function, which outputs HTML augmented with the HTML-comment demarcations to denote blocks.

Both `pasteHandler` and `serialize` come from the `@wordpress/blocks` module, which is sometimes known as the _blocks _API_.

File: packages/block-editor/src/components/copy-handler/README.md [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+## Table of contents
+
+1. [Development guidelines](#development-guidelines)
+2. [Related components](#related-components)
+
+## Development guidelines
+
+### Usage
+
+Adds the copy handler functionnalities to the editor.
```
Comment (mcsf) [inline packages/block-editor/src/components/copy-handler/README.md:18]:
```suggestion
Adds block-level copy-and-paste support to the editor.
```

Comment (mcsf) [reply packages/block-editor/src/components/copy-handler/README.md:18]: It's important that this distinction be clear, as `CopyHandler` doesn't handle (and isn't required for) copying and pasting of fragments of content within a block. Those will be natively handled (e.g. selecting a piece of the block UI and copying, or selecting a fragment of RichText and copying).

File: packages/block-editor/src/components/copy-handler/README.md [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+import { BlockList, CopyHandler } from '@wordpress/block-editor';
+
+const MyCopyHandler = () => (
+	<CopyHandler>
+		<BlockList />
+	</CopyHandler>
+);
+```
+
+_Note:_ The `CopyHandler` only catches cut, copy and paste events coming from its props.children. In this example, the child is the `BlockList` component.
```
Comment (mcsf) [inline packages/block-editor/src/components/copy-handler/README.md:30]: `CopyHandler` also only intercepts such events when there is a block-editing context and at least one block is currently selected.

File: packages/block-editor/src/components/copy-handler/README.md [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+
+### Props
+
+### `children`
+
+-  Type: `Element`
+
+Children are passed as children of `CopyHandler`.
```
Comment (mcsf) [inline packages/block-editor/src/components/copy-handler/README.md:38]: This sentence is a bit redundant. :) 

```suggestion
The elements to be rendered and whose `cut`, `copy` and `paste` events we'd like to intercept.
```

**Review by mcsf — CHANGES_REQUESTED** [review]

**Comment by JustinyAhin** [comment]
Thank your for deep looking into this @mcsf. Just added your changes requests in a new commit.

**Review by mcsf — APPROVED** [review]
Thanks for this PR!

**Pull request merged**